include/linker/devicetree_regions.h

```diff
@@ -7,6 +7,34 @@
  * Generate memory regions from devicetree nodes.
  */
 
+/**
+ * @brief Get the linker memory-region name
+ *
+ * This attempts to use the zephyr,linker-region property, falling back
+ * to the node path if it doesn't exist.
+ *
+ * Example devicetree fragment:
+ *
+ *     / {
+ *             soc {
+ *                     sram1: memory@2000000 {
+ *                         zephyr,linker-region = "MY_NAME";
+ *                     };
+ *                     sram2: memory@2001000 { ... };
+ *             };
+ *     };
+ *
+ * Example usage:
+ *
+ *    LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sram1)) // "MY_NAME"
+ *    LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sram2)) // "/soc/memory@2001000"
+ *
+ * @param node_id node identifier
+ * @return the name of the linker region the node will generate
+ */
+#define LINKER_DT_NODE_REGION_NAME(node_id) \
+	DT_PROP_OR(node_id, zephyr_linker_region, DT_NODE_PATH(node_id))
```

Comment:
We should have same behavior here, so that region names are picked up from devicetree if defined regardless of how the linker script is generated.
https://github.com/zephyrproject-rtos/zephyr/blob/513517fa5c913fc6bb9e6fac32a3368dea8782b6/cmake/linker_script/arm/linker.cmake#L37-L50

which would be in this function where name from region property used, otherwise fallback on node path.
https://github.com/zephyrproject-rtos/zephyr/blob/513517fa5c913fc6bb9e6fac32a3368dea8782b6/cmake/extensions.cmake#L3109

Reply:
done